### ISSUE TITLE
PlasmaComponents3.ComboBox doesnt highlight items in Adapta theme

### ISSUE THREAD

**Issue by psifidotos** [PapirusDevelopmentTeam/adapta-kde#26]
![default](https://user-images.githubusercontent.com/2216474/34005980-6666d74a-e105-11e7-83be-7dd1816b6bc4.png)

this issue doesnt occur with Breeze themes:

![default](https://user-images.githubusercontent.com/2216474/34006077-bf344bf0-e105-11e7-86c1-fb174bd49b23.png)

**Comment by varlesh** [comment]
I think it's upstream KDE or Kvantum bug. Maybe style widget harcoded, it's possible for some plasma elements.
This reproduced on other (original) Kvantum themes. For example, default:
![image](https://user-images.githubusercontent.com/8083855/34011916-3cc271d0-e12b-11e7-9ff6-29d7a22b006f.png)

**Comment by varlesh** [comment]
@psifidotos Sorry, i'm not programmer. Maybe @tsujan help you.
Not forgot some elements on plasma hardcoded.
See examples:
![image](https://user-images.githubusercontent.com/8083855/34012442-2fedf810-e12d-11e7-81a2-7cc4a610780c.png)
![image](https://user-images.githubusercontent.com/8083855/34012466-4a5409a6-e12d-11e7-9e10-4e01015754bd.png)

**Comment by varlesh** [comment]
This l"menu list" used from WM engine style with wrong color-scheme... and i don't know why.
See example:

- engine - Kvantum
- kvantum theme - Kvantum (default)
- plasma theme - Breeze
- colorscheme - Kvantum

![image](https://user-images.githubusercontent.com/8083855/34012687-2b996690-e12e-11e7-88a5-dd4d110af24e.png)

**Comment by varlesh** [comment]
@tsujan

**Comment by varlesh** [comment]
I think better solution use plasma widget background and style for this list menu. See on plasma this used for panel:
![image](https://user-images.githubusercontent.com/8083855/34012905-f4a0df46-e12e-11e7-8fee-29b89a5b83ab.png)

**Comment by tsujan** [comment]
@varlesh Please tell me what that window is -- I don't know Russian and can't find a similar window/dialog here.

**Comment by varlesh** [comment]
@tsujan It's [latte-dock](https://github.com/psifidotos/Latte-Dock) list menu

**Comment by tsujan** [comment]
> It's latte-dock list menu

OK, thanks! I'll investigate it. Can you see a similar issue elsewhere?

**Comment by varlesh** [comment]
@psifidotos Yes i see this bug for other list menu on kde systemsettings.... but i'm forgot where))) It was a long time ago

**Comment by tsujan** [comment]
> Yes i see this bug for other list menu on kde systemsettings.... but i'm forgot where))) It was a long time ago

Tell me whenever you remember it!

I think this should be about a problem in QML -- not in Kvantum -- but if I find the cause, I'll try to include a workaround in Kvantum (in addition to other QML workarounds that are already there).

**Comment by omni6** [comment]
:)

**Comment by tsujan** [comment]
I installed latte-dock 0.7.2 (from AUR, which gets it from https://download.kde.org/stable/latte-dock/) and found no problem with KvAdaptaDark:

![latte-dock](https://user-images.githubusercontent.com/981076/34014276-d66d9b8a-e12f-11e7-82d5-d64629cd2da3.png)

The checked row is always highlighted (a QML peculiarity) but other rows are also highlighted on mouse-over.

**Comment by psifidotos** [comment]
Dont use Latte Dock to debug because in Latte Dock I havent updated yet to PlasmaComponent3.ComboBox

you can use: 
[highlight.qml.txt](https://github.com/PapirusDevelopmentTeam/adapta-kde/files/1560735/highlight.qml.txt)

1. when you download it  rename it to `highlight.qml`
2. just run in that directory the command `qmlscene highlight.qml`
3. the first ComboBox is a PlasmaComponents2.ComboBox and the second is a PlasmaComponents3.ComboBox that shows the issue.

**Comment by tsujan** [comment]
@varlesh I can't see any problem in QML combo menus belonging to KDE. The above-mentioned latte-dock, which I installed, is from KDE. The problem you see can be in https://github.com/psifidotos/Latte-Dock.

**Comment by psifidotos** [comment]
> I can't see any problem in QML combo menus belonging to KDE

the `highlight.qml` file works ok for you?

>  The problem you see can be in https://github.com/psifidotos/Latte-Dock.

in what way?

**Comment by tsujan** [comment]
> in what way?

I don't know. As as told above, I installed latte-dock from AUR (Arch) and AUR gets it from https://download.kde.org/stable/latte-dock/. I don't see any issue in other combo menus either.

**Comment by psifidotos** [comment]
@tsuhan can you show me a screenshot of running `highlight.qml` in your system following the above commands and opening the second combobox?

**Comment by tsujan** [comment]
@psifidotos I don't know what you're talking about (what's highlight.qml?). I attached a screenshot from latte-dock [above](https://github.com/PapirusDevelopmentTeam/adapta-kde/issues/26#issuecomment-351836025).

**Comment by psifidotos** [comment]
@tsuhan

you can use: 
[highlight.qml.txt](https://github.com/PapirusDevelopmentTeam/adapta-kde/files/1560735/highlight.qml.txt)

1. when you download it  rename it to `highlight.qml`
2. just run in that directory the command `qmlscene highlight.qml`
3. the first ComboBox is a PlasmaComponents2.ComboBox and the second is a PlasmaComponents3.ComboBox that shows the issue.

BTW the plasma systemsettings havent all been updated to PlasmaComponents2 or PlasmaComponents3 which are qml yet... most of them at Qt Widgets... Your screenshot is from the Latte plasmoid and not from the Latte App... You can run Latte from the app launcher and afterwards with right click to access its settings, but first try the above steps... PlasmaComponents3 is going to be the next step to plasma and most of plasma systemsettings in the future will include PlasmaComponent3 and kirigami qml code.

**Comment by psifidotos** [comment]
> @varlesh I think it's upstream KDE or Kvantum bug. Maybe style widget harcoded, it's possible for some plasma elements.

concerning KDE I dont think so because plasma default themes dont have such issues. You can debug in your system by comparing with plasma themes that are very different in coloring such as Breeze and Breeze-Dark

**Comment by tsujan** [comment]
@psifidotos 
As I said twice above, I installed from AUR -- not from https://github.com/psifidotos/Latte-Dock -- because I'm very busy with several projects. This is plasma-5.11.4 and *all* QML combo menus are OK. If I see just one example of a malformed QML combo menu, I'll investigate the problem. That's why I said the issue *can* be in https://github.com/psifidotos/Latte-Dock (but not necessarily).

**Comment by tsujan** [comment]
By "seeing just one example of a malformed QML combo menu", I meant a combo menu *other than* that of https://github.com/psifidotos/Latte-Dock.

**Comment by psifidotos** [comment]
@tsujan no prob. you will see them in the near future...

**Comment by tsujan** [comment]
BTW, a question: Ahat's the relation between https://download.kde.org/stable/latte-dock/ and https://github.com/psifidotos/Latte-Dock? Are they the same program? If so, why are they so different? If not, why do they have the same name?

**Comment by psifidotos** [comment]
they are exactly the same program :) and I am the maintainer

> Are they the same program? If so, why are they so different?

in what way are they different?

**Comment by psifidotos** [comment]
btw, I love kvantum... :) it made my system totally awesome!!!

**Comment by tsujan** [comment]
>they are exactly the same program :) and I am the maintainer

Thanks for the info! I didn't know that.

> in what way are they different?

(1) My screenshot (from AUR) is very different from yours.
(2) The latest release in https://github.com/psifidotos/Latte-Dock is 0.7.1, while the latest one in https://download.kde.org/stable/latte-dock/ is 0.7.2.

If you really think the issue is in Kvantum, please open an issue at https://github.com/tsujan/Kvantum! I might find some free time to look into the code of Latte-Dock later but I'm a C++ coder -- QML isn't my area of expertise. Therefore, I should wait until I find another instance of malformed combo menu.

**Comment by tsujan** [comment]
OK, now I understand what the problem is: English ;)

I logged into LXQt (my main DE), saw latte-dock in the start menu, clicked on its icon and it showed up at the bottom of the desktop. Very nice dock! It's usable under LXQt too :)

Now, I can see what you see :) However, I'm still not sure about the cause.

**Comment by psifidotos** [comment]
> OK, now I understand what the problem is: English ;)

no worries

> (1) My screenshot (from AUR) is very different from yours.

just run in the terminal the command `latte-dock` 

>  The latest release in https://github.com/psifidotos/Latte-Dock is 0.7.1, , while the latest one in https://download.kde.org/stable/latte-dock/ is 0.7.2.

Latte was first appeared in github but there was much demand from many users in different distros so we made it an official kde project... to celebrate this the 0.7.2 version and everyone after that will be published totally through kde infrastructure. :)

concerning the issue I managed to find a fix, the result is the following when using in Latte PlasmaComponents3.ComboBox : 

![default](https://user-images.githubusercontent.com/2216474/34016964-7d101fd4-e12c-11e7-9196-8e9eac99a2c4.png)

to achieve that I replaced the file `plasma/desktoptheme/Adapta/widgets/listitem.svg` with the one from the Breeze plasma theme. 


The only way to see ALL the combobox issues at the same time is to use the `highlight.qml` file that I sent you earlier... if you want help to do that I am here to help... ;)

**Comment by tsujan** [comment]
I'm pretty sure there's a styling problem in latte-dock's code. For example, sliders aren't drawn correctly:

![latte-dock1](https://user-images.githubusercontent.com/981076/34017125-a2132972-e139-11e7-9e6b-f71f3e5c7b9d.png)

Such problems are about incomplete hard-coded styles. Sorry that I can't read its code for now!

**Comment by psifidotos** [comment]
> I'm pretty sure there's a styling problem in latte-dock's code. For example, sliders aren't drawn correctly

Latte doesnt style anything :) we dont use any Latte-hardcoded specific colors and we dont draw anything also, everything is provided by plasma libraries... try to open Latte under a plasma environment to see the difference...

**Comment by tsujan** [comment]
> everything is provided by plasma libraries

*Somewhere*, there should be a wrong hard-coded style -- I say that based on experience. If you know where the styling is done, please tell me the name of the source file (XXX.cpp)!

The styling problem aside, I love this dock! Started to use it under LXQt:

![latte-dock2](https://user-images.githubusercontent.com/981076/34017361-b8c790c6-e13a-11e7-8f47-45ef44c66a97.png)

**Comment by psifidotos** [comment]
the library is called  plasma-framework and it is an official kde framework: https://github.com/KDE/plasma-framework

the PlasmaComponents2 that Latte is using are found at: https://github.com/KDE/plasma-framework/tree/master/src/declarativeimports/plasmacomponents/qml

the PlasmaComponents3 which are going to replace them and they are the next-gen and one to be used in the future from plasma both for desktop and mobile are found at: https://github.com/KDE/plasma-framework/tree/master/src/declarativeimports/plasmacomponents3

but these are qml files... their drawing is set by the plasma theme which is svg files ... For Adapta that would be: https://github.com/PapirusDevelopmentTeam/adapta-kde/tree/master/plasma/desktoptheme/Adapta/widgets

a plasma theme contains only a color palette and svg files... is not using any other drawing... at least these are my info :)

in the `widgets` directory there are .svg files for everything...

**Comment by psifidotos** [comment]
@tsuhan what plasma theme are you using in order to check it in my system? I havent seen that theme before...

**Comment by psifidotos** [comment]
@tsujan If you want to take a look what Latte brought to the users you can watch: https://www.youtube.com/watch?v=74EA3nijbTU  
the top bar in the video is also a Latte dock/panel... ;)

**Comment by tsujan** [comment]
> try to open Latte under a plasma environment to see the difference...

No difference here. I logged into plasma KDE and took this screenshot a minute ago:

![latte-dock3](https://user-images.githubusercontent.com/981076/34017863-a3f1429e-e13c-11e7-954f-afd4409514aa.png)

> what plasma theme are you using in order to check it in my system?

I don't think the theme should make any difference but this is an old SVG theme I have under KDE: https://www.dropbox.com/s/w3ovywtfn51ffe7/TranslucyDark.tar.xz

If latte-dock takes its style from Plasma completely (as you mentioned), then the problem should be in Plasma. However, the question arises as to why such a problem can't be seen in other places.

Anyhow, latte-dock is a very nice app. Well done! Keep up the good work!

**Comment by tsujan** [comment]
Just for the record, Plasma (QML) combo menus are all like the following screenshot (task manager settings with a dark Kvantum theme) and are quite readable with dark themes:

![task_manager](https://user-images.githubusercontent.com/981076/34018415-0231b5a8-e13f-11e7-8407-3b4711d4b906.png)

If they're going to be changed and become like the combo menu of latte-dock, we should wait for a while, I guess.

**Comment by tsujan** [comment]
And the same for sliders:

![slider](https://user-images.githubusercontent.com/981076/34018544-8fcb087e-e13f-11e7-8946-39dc11f99d22.png)

**Comment by psifidotos** [comment]
> However, the question arises as to why such a problem can't be seen in other places. 

Plasma is using two themes... for its settings windows is using a different way of painting (this is where kvantum takes place) than its widgets(plasmoids) and panels. For them uses the plasma theme. The thing is that there arent many qml plamoids or panel settings using ComboBoxes or the Sliders issues in order to indicate the inconsistencies are fixes need in the plasma theme...

The sliders issue can be easily observed in plasma. In the plasma systray there is a volume control. If you open it you will see that the sliders in there are shown exactly like Latte.

![default](https://user-images.githubusercontent.com/2216474/34019879-a34d47aa-e139-11e7-8d6b-b830bf24d273.png)

**Comment by tsujan** [comment]
@psifidotos I know that TranslucyDark's sliders don't have a trough (which is an issue in the theme). My point is quite different. Let me explain it with more details:

All other KDE settings dialogs take their widget styles from the active style engine -- Kvantum, in this case -- but latte-dock takes it from the Plasma theme. Let's suppose that this is the future of Plasma, i.e. all other settings will become like latte-dock's at some point. So far, so good! However -- and this is my main argument -- IF PLASMA IS USED FOR STYLING, THEN THE ACTIVE STYLE PLUGIN SHOULD **NOT** BE CONSULTED; otherwise, a mess will be created.

Now, if latte-dock's settings dialog is styled by Plasma, its combo menu should be completely styled by plasma too. Apparently, that isn't the case. This is what I call *an inconsistent/incomplete hard-coded style*.

**Comment by tsujan** [comment]
Ah, this discussion was based on a wrong information. Why did you say it's specific to Kvantum? This happens with Breeze too:

![breeze](https://user-images.githubusercontent.com/981076/34021400-e2a81cb2-e14e-11e7-875a-8d265a285e65.png)

**Comment by varlesh** [comment]
> Why did you say it's specific to Kvantum?

It's i'm say. Sorry guys i'm not programmer and i don't know all subtleties.

Anyway this list menu used from WM engine (Why?) with wrong style. I think it's plasma bug...

**Comment by tsujan** [comment]
> Sorry guys i'm not programmer

No need to be sorry. It's good to know the cause of a problem :)

> I think it's plasma bug

Definitely.

**Comment by psifidotos** [comment]
> IF PLASMA IS USED FOR STYLING, THEN THE ACTIVE STYLE PLUGIN SHOULD NOT BE CONSULTED

yeah you are right, I opened a bug for **PlasmaComponents2.ComboBox** let's see what they will reply: https://bugs.kde.org/show_bug.cgi?id=387922

even though the plasma devs have moved their focus to **PlasmaComponents3** for which the **PlasmaComponents3.ComboBox fixes this issue**...

so my question is:

Do you want to see a **PlasmaComponents3.ComboBox** and fix in Adapta its appearance? (Latte doesnt have yet a **PlasmaComponents3.ComboBox** in its settings the screenshot in the start is from a Latte development version in my system!!)

**Comment by tsujan** [comment]
> Do you want to see a PlasmaComponents3.ComboBox and fix in Adapta its appearance?

You mean the Adapta *Plasma* theme? Because the Adapta Kvantum theme doesn't have a role here.

**Comment by varlesh** [comment]
@psifidotos This reproduced on Breeze plasma theme too

**Comment by psifidotos** [comment]
> You mean the Adapta Plasma theme?

yep... and more specific that file which has the problem: https://github.com/PapirusDevelopmentTeam/adapta-kde/blob/master/plasma/desktoptheme/Adapta/widgets/listitem.svg

> @psifidotos This reproduced on Breeze plasma theme too

no it isnt...
Because you havent seen a **PlasmaComponents3.ComboBox** yet, Latte isnt using yet...
so the question one more time is:
Do you want to see **in your system** a _PlasmaComponents**3**.ComboBox_ how it looks with Adapta plasma theme?

**Comment by tsujan** [comment]
@psifidotos 
I think varlesh meant that the problem with the combo menu of latte-dock also happens with Breeze Plasma theme. My last screenshot [above](https://user-images.githubusercontent.com/981076/34021400-e2a81cb2-e14e-11e7-875a-8d265a285e65.png) was taken with the Breeze style engine + the dark Breeze Plasma theme. So, if Breeze has problem, what could be done in Adapta?

**Comment by psifidotos** [comment]
**A) PlasmaComponents2.ComboBox :**
![default](https://user-images.githubusercontent.com/2216474/34045786-1bb3a512-e1b3-11e7-83c8-5cae3e598bbd.png)

**B) PlasmaComponents3.ComboBox**
![default](https://user-images.githubusercontent.com/2216474/34045794-2dc9e9aa-e1b3-11e7-99a0-9a39397a56ba.png)


**For (A) you cant do anything it is a plasma bug and I reported it...
For (B) it is Adapta plasma theme bug and yes you can do...**

**So again: Do you want to observe (B) in your system?  Because Latte isnt the way to observe (B). Latte offers only (A) for the time being...**

**Comment by varlesh** [comment]
@psifidotos I don't know how fix that. 
Adapta Plasma theme contained only SVG-files and colors (from system). it's all! No scripts, no hooks and etc...

**Comment by psifidotos** [comment]
@varlesh I fixed it by replacing the Adapta file : desktoptheme/Adapta/widgets/listitem.svg with the one from Breeze theme, it looks like this after that:

![default](https://user-images.githubusercontent.com/2216474/34047702-bcb8a93e-e1b9-11e7-9e08-23bd444cea2d.png)

Why this is happening?

the Breeze svg file contains css code in it if you open it... and the colors used are based on that css... The plasma theme engine possibly uses that css in order to substitute that colors with the colors used from the current plasma theme pallete...

**Comment by varlesh** [comment]
Adapta not support CSS and KDE Color scheme, because it's spoil original style. Also it's not obligatory....
 [Arc Color](https://github.com/PapirusDevelopmentTeam/arc-kde) support CSS. you check this if you want

**Comment by psifidotos** [comment]
> Arc Color support CSS. you check this if you want

I tried Arc Dark and it works just fine...

> Adapta not support CSS and KDE Color scheme, because it's spoil original style.

no idea, feel free to respond any how you please, I just wanted to report an issue... PlasmaComponents3.ComboBox and Adapta just dont love each other... that was the issue... ;)
